Mathlib/RingTheory/Derivation/MapCoeffs.lean

```diff
@@ -99,3 +101,117 @@ theorem apply_eval_eq (x : A) (p : A[X]) :
   apply_aeval_eq d x p
 
 end Derivation
+
+namespace Differential
+
+variable {A : Type*} [CommRing A] [Differential A]
+
+/--
+A specialization of `Derivation.mapCoeffs` for the case of a differential ring.
+-/
+def mapCoeffs : Derivation ℤ A[X] A[X] :=
+  PolynomialModule.equivPolynomialSelf.compDer Differential.deriv.mapCoeffs
+
+@[simp]
+lemma mapCoeffs_apply (p : A[X]) (i) :
+    coeff (mapCoeffs p) i = (coeff p i)′ := rfl
+
+@[simp]
+lemma mapCoeffs_monomial (n : ℕ) (x : A) :
+    mapCoeffs (monomial n x) = monomial n x′ := by
+  simp [mapCoeffs]
+
+@[simp]
+lemma mapCoeffs_X :
+    mapCoeffs (X : A[X]) = 0 := by simp [← monomial_one_one_eq_X]
+
+@[simp]
+lemma mapCoeffs_C (x : A) :
+    mapCoeffs (C x) = C x′ := by simp [← monomial_zero_left]
+
+variable {R : Type*} [CommRing R] [Differential R] [Algebra A R] [DifferentialAlgebra A R]
+
+theorem deriv_aeval_eq (x : R) (p : A[X]) :
+    (aeval x p)′ = aeval x (mapCoeffs p) + aeval x (derivative p) * x′ := by
+  convert Derivation.apply_aeval_eq' Differential.deriv _ (Algebra.linearMap A R) ..
+  · simp [mapCoeffs]
+  · simp [deriv_algebraMap]
+
+/--
+The unique derivation which can be made to a `DifferentialAlgebra` on `A[X]` with
+`X′ = v`.
+-/
+def implicitDeriv (v : A[X]) :
+    Derivation ℤ A[X] A[X] :=
+  mapCoeffs + v • derivative'.restrictScalars ℤ
+
+@[simp]
+lemma implicitDeriv_C (v : A[X]) (b : A) :
+    implicitDeriv v (C b) = C b′ := by
+  simp [implicitDeriv]
+
+@[simp]
+lemma implicitDeriv_X (v : A[X]) :
+    implicitDeriv v X = v := by
+  simp [implicitDeriv]
+
+/--
+`implicitDeriv` as a `Differential`.
+-/
+abbrev implicitDifferential (v : A[X]) : Differential A[X] := ⟨implicitDeriv v⟩
+
+lemma implicitDifferentialAlgebra (v : A[X]) :
+    letI := implicitDifferential v
+    DifferentialAlgebra A A[X] :=
+  letI := implicitDifferential v
+  ⟨implicitDeriv_C v⟩
+
+@[simp]
+lemma implicitDifferential_C (v : A[X]) (b : A) :
+    letI := implicitDifferential v
+    (C b)′ = C b′ := implicitDeriv_C v b
+
+@[simp]
+lemma implicitDifferential_X (v : A[X]) :
+    letI := implicitDifferential v
+    X′ = v := implicitDeriv_X v
```

Comment:
I suggest leaving this block out of this PR, until you need them for something. The `letI` in the statements suggests to me that these will be somewhat tricky to use, and we might converge onto a more usable design late.

Reply:
Alright, I'll add this in a later PR